libcst/matchers/_visitors.py

```diff
@@ -58,11 +65,11 @@ def _get_possible_match_classes(matcher: BaseMatcherNode) -> List[Type[cst.CSTNo
         return [getattr(cst, matcher.__class__.__name__)]
 
 
-def _get_possible_annotated_classes(annotation: Type[object]) -> List[Type[object]]:
+def _get_possible_annotated_classes(annotation: object) -> List[Type[object]]:
     if getattr(annotation, "__origin__", None) is Union:
         return getattr(annotation, "__args__", [])
     else:
-        return [annotation]
+        return [cast(Type[object], annotation)]
```

Comment:
We could add runtime assertions that there isn't a nested Union[Union[...]] or anything else weird, but this seems like overkill for a best-effort runtime validation.